### ISSUE TITLE
Database smoke tests failing on Tier 3 SRE

### ISSUE THREAD

**Issue by craddm** [alan-turing-institute/data-safe-haven#2123]
## :white_check_mark: Checklist

- [x] I have searched open and closed issues for duplicates.
- [x] This is a problem observed when **deploying** a Data Safe Haven.
- [x] I can reproduce this with the [latest version](https://github.com/alan-turing-institute/data-safe-haven/releases).
- [x] I have read through the [documentation](https://alan-turing-institute.github.io/data-safe-haven/).
- [x] This isn't an open-ended question (open a [discussion](https://github.com/alan-turing-institute/data-safe-haven/discussions) if it is).

## :computer: System information

- **Operating System:** macOS
- **Data Safe Haven version:** `update-nexus-allowlist` [3035d81](https://github.com/craddm/data-safe-haven/commit/3035d811abd00649ed90758f3861c30b76efa49a) (parity with `release-v5.0.0`)

## :package: Packages

<details>
<summary>List of packages</summary>

```none
Paste list of packages here
```

</details>

## :no_entry_sign: Describe the problem

Smoke tests involving databases are failing for both Python and R

## :deciduous_tree: Log messages

<details>
<summary>Relevant log messages</summary>

```none
 root@shm-green-sre-noteasybeing-vm-workspace-01:/usr/local/smoke_tests# bats run_all_tests.bats -f "database"
 ✗ MS SQL database (Python)
   (in test file run_all_tests.bats, line 101)
     `[ "$status" -eq 0 ]' failed
 ✗ MS SQL database (R)
   (in test file run_all_tests.bats, line 111)
     `[ "$status" -eq 0 ]' failed
 ✗ Postgres database (Python)
   (in test file run_all_tests.bats, line 119)
     `[ "$status" -eq 0 ]' failed
```

</details>

**Comment by jemrobinson** [comment]
I just got this result from 3035d81 when running the smoke tests as `root`:

```console
root@shm-pink-sre-fuschia-vm-workspace-02:/usr/local/smoke_tests# bats run_all_tests.bats 
 ✓ Mounted drives (/data) 
 ✓ Mounted drives (/home) 
 ✓ Mounted drives (/output) 
 ✓ Mounted drives (/shared) 
 ✓ Python package repository 
 ✓ R package repository 
 ✓ Python functionality 
 ✓ R functionality 
 ✓ MS SQL database (Python) 
 ✓ MS SQL database (R) 
 ✓ Postgres database (Python) 
 ✓ Postgres database (R) 

12 tests, 0 failures
```

**Comment by jemrobinson** [comment]
Is this the issue (in `test_databases.sh`):

https://github.com/alan-turing-institute/data-safe-haven/blob/develop/data_safe_haven/resources/workspace/ansible/files/usr/local/smoke_tests/test_databases.sh#L25-L30

we make this assumption about the database hostname

```
sre_prefix="$(hostname | cut -d "-" -f 1-4)"
hostname="${sre_prefix}-db-server-mssql"
```

while we should probably pass this in as a variable.

**Comment by jemrobinson** [comment]
It's not this - it's incorrect escaping when we write special characters to `/etc/database_credential`.

**Comment by craddm** [comment]
Fixed by #2125
